JBBCode/Parser.php

```diff
@@ -97,7 +97,7 @@ public function addBBCode($tagName, $replace, $useOption = false, $parseContent
      */
     public function addCodeDefinition(CodeDefinition $definition)
     {
-        $this->bbcodes[] = $definition;
+        $this->bbcodes[$this->getTagKey($definition->getTagName(), $definition->usesOption())] = $definition;
         return $this;
```

Comment:
you could always try this, as well:

```
$tagName = $definition->getTagname();
if(!isset($this->bbcodes[$tagName]){
   $this->bbcodes[$tagName] = [];
}
$this->bbcodes[$tagName][$definition->usesOption()] = $definition;
```

Then to look things up requires no serialization, JSON-ization, etc:

```
public function getCode($tagName, $usesOption = false){
   if($this->codeExists($tagName, $usesOption)) {
            return $this->bbcodes[$tagName][$usesOption];
   }
}
```